### PR TITLE
Changeset query endpoint

### PR DESCRIPTION
**Pull request by dereklieu**
Adds `/api/0.6/changesets?user=USER_ID`

Also refactors error handling.

This is to allow for an upcoming push that bundles current iD.

### CODE REVIEW

File: routes/changeset-create.js [hunk 1]
```diff
@@ -1,82 +1,72 @@
...
-  req.payload.user = req.payload.user || 'placeholder';
-
   var changeset = req.payload.osm.changeset;
   changeset.tag = validateArray(changeset.tag);
-  var uid = req.payload.uid;
-  var userName = req.payload.user;
-
-  if (!uid || !userName) {
-    return res(Boom.badRequest('A new changeset must include a user id and a username.'));
-  }
```
Comment (dereklieu) [inline routes/changeset-create.js:21]: We're already checking and defaulting, this error would never happen.

File: routes/changeset-query.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
+'use strict';
```
Comment (dereklieu) [inline routes/changeset-query.js:1]: For reference, [OSM implementation](https://github.com/openstreetmap/openstreetmap-website/blob/master/app/controllers/changeset_controller.rb#L204)

**Comment by dereklieu** [comment]
@kamicut gonna go ahead and merge this so we can get a post out. Let me know if you have any questions though.

**Comment by kamicut** [comment]
👍  sorry couldn't get to it earlier @dereklieu

**Pull request merged**